autoprecompiles/src/basic_blocks.rs

```diff
@@ -0,0 +1,112 @@
+use std::collections::BTreeSet;
+
+use serde::{Deserialize, Serialize};
+
+use crate::SymbolicInstructionStatement;
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct SymbolicBlock<T> {
```

Comment:
What is symbolic about all these concepts? Isn't all we have about it concrete?

Reply:
I started from `SymbolicInstructionStatement` and wrapped it, so I kept the `Symbolic` prefix. More than happy to rename to `BasicBlock` and `Program`